equinix/resource_metal_connection.go

```diff
@@ -340,9 +343,59 @@ func resourceMetalConnectionUpdate(d *schema.ResourceData, meta interface{}) err
 			return friendlyError(err)
 		}
 	}
+
+	// Don't update VLANs until _after_ the main ConnectionUpdateRequest has succeeded
+	if d.HasChange("vlans") {
+		connType := packngo.ConnectionType(d.Get("type").(string))
+
+		if connType == packngo.ConnectionShared {
+			old, new := d.GetChange("vlans")
```

Comment:
@ocobleseqx @displague I have a local commit that updates this code to throw an error if `status` is `pending`, but that change causes the acceptance test to fail.  Without the `status` check, the acceptance test passes, so I _think_ that indicates that the API is correctly accepting the VLANs updates for `pending` shared connections.  I'm hesitant to update the tests to handle the `status` check if everything is working without the `status` check.

Reply:
I checked it again using the API and indeed if it can be updated even it is still `pending`, I don't know what was wrong with my prev tests